src-electron/generator/helper-zcl.js

```diff
@@ -759,28 +767,37 @@ function zcl_attributes(options) {
  * @param {*} options
  * @returns Promise of attribute iteration.
  */
-function zcl_attributes_client(options) {
+async function zcl_attributes_client(options) {
   // If used at the toplevel, 'this' is the toplevel context object.
   // when used at the cluster level, 'this' is a cluster
-  let promise = templateUtil
-    .ensureZclPackageIds(this)
-    .then((packageIds) => {
-      if ('id' in this) {
-        return queryZcl.selectAttributesByClusterIdAndSideIncludingGlobal(
-          this.global.db,
-          this.id,
-          packageIds,
-          dbEnum.side.client
-        )
-      } else {
-        return queryZcl.selectAllAttributesBySide(
-          this.global.db,
-          dbEnum.side.client,
-          packageIds
-        )
-      }
-    })
-    .then((atts) => templateUtil.collectBlocks(atts, options, this))
+  let packageIds = await templateUtil.ensureZclPackageIds(this)
+  let clientAttributes = ''
+  if ('id' in this) {
+    // We're functioning inside a nested context with an id, so we will only query for this cluster.
+    clientAttributes =
+      await queryZcl.selectAttributesByClusterIdAndSideIncludingGlobal(
+        this.global.db,
+        this.id,
+        packageIds,
+        dbEnum.side.client
+      )
+    clientAttributes = await upgrade.computeStorageTemplate(
+      this.global.db,
+      this.id,
+      clientAttributes
+    )
+  } else {
+    clientAttributes = await queryZcl.selectAllAttributesBySide(
+      this.global.db,
+      dbEnum.side.client,
+      packageIds
+    )
+  }
+  if ('removeKeys' in options.hash) {
```

Comment:
Again, this is adding removeKeys to something that didn't use to have it, right?

Reply:
yeah, I just figured since it was being used for zcl_attributes_server that I would include it and let someone tell me it was not needed. Thank you, will remove.